benchexec/tools/fusebmc.py

```diff
@@ -5,41 +5,50 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-import benchexec.tools.esbmc as esbmc
-from benchexec.tools.template import ToolNotFoundException
+import os
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+import benchexec.tools.template
+import benchexec.result as result
 
-
-class Tool(esbmc.Tool):
+class Tool(benchexec.tools.template.BaseTool2):
     """
     This class serves as tool adaptor for FuSeBMC (https://github.com/kaled-alshmrany/FuSeBMC)
     """
 
-    REQUIRED_PATHS_TESTCOMP20 = ["esbmc", "esbmc-wrapper.py", "my_instrument"]
-    REQUIRED_PATHS_TESTCOMP21 = [
-        "esbmc",
-        "fusebmc.py",
-        "FuSeBMC_inustrment/FuSeBMC_inustrment",
-        "fusebmc_output",
-        "map2check-fuzzer",
-    ]
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("fusebmc.py")
+
+    def working_directory(self, executable):
```

Comment:
I wonder why setting the working directory is now necessary when it was not necessary before?

Note that requiring a specific working directory is not a problem for BenchExec, but might be inconvenient for your users.

Reply:
I guess because before was inherited from esbmc, and now we want it to be independent of esbmc.

Reply:
Right, I overlooked this.